client/src/app/queries/businessservices.ts

```diff
@@ -1,45 +1,75 @@
 import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { AxiosError } from "axios";
 
 import {
+  createBusinessService,
   deleteBusinessService,
   getBusinessServiceById,
   getBusinessServices,
+  updateBusinessService,
 } from "@app/api/rest";
-import { BusinessService } from "@app/api/models";
-import { AxiosError } from "axios";
 
 export const BusinessServicesQueryKey = "businessservices";
 export const BusinessServiceQueryKey = "businessservice";
 
 export const useFetchBusinessServices = () => {
-  const { data, isLoading, error, refetch } = useQuery<BusinessService[]>(
-    [BusinessServicesQueryKey],
-    async () => (await getBusinessServices()).data,
-    {
-      onError: (error) => console.log("error, ", error),
-    }
-  );
+  const { data, isLoading, error, refetch } = useQuery({
+    queryKey: [BusinessServicesQueryKey],
+    queryFn: getBusinessServices,
+    onError: (error: AxiosError) => console.log("error, ", error),
+  });
   return {
     businessServices: data || [],
     isFetching: isLoading,
     fetchError: error as AxiosError,
     refetch,
   };
 };
-export const useFetchBusinessServiceByID = (id: number | string) => {
-  const { data, isLoading, error } = useQuery<BusinessService>(
-    [BusinessServicesQueryKey, id],
-    async () => (await getBusinessServiceById(id)).data,
-    { onError: (error) => console.log(error) }
-  );
 
+export const useFetchBusinessServiceByID = (id: number | string) => {
+  const { data, isLoading, error } = useQuery({
+    queryKey: [BusinessServicesQueryKey, id],
+    queryFn: () => getBusinessServiceById(id),
+    onError: (error: AxiosError) => console.log("error, ", error),
+  });
   return {
     businessService: data,
     isFetching: isLoading,
     fetchError: error as AxiosError,
   };
 };
 
+export const useCreateBusinessServiceMutation = (
+  onSuccess: (res: any) => void,
+  onError: (err: AxiosError) => void
+) => {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: createBusinessService,
+    onSuccess: (res) => {
+      onSuccess(res);
+      queryClient.invalidateQueries([BusinessServicesQueryKey]);
+    },
+    onError,
+  });
+};
+
+export const useUpdateBusinessServiceMutation = (
+  onSuccess: (res: any) => void,
```

Comment:
```suggestion
  onSuccess: (businessServiceName: string) => void,
```

Reply:
@ibolton336, all queries are using `res` as parameter for onSuccess().
Therefore I don't see the value of such change and the gain in the meaning.
I'm not necessarily against it meanwhile if we decide such change then we'd have to make it consistent across the all queries.

Reply:
@ibolton336, you had a good point with the `any` type.
This isn't the case anymore as there is no need to pass anything for now (see update) until we eventually address that later if we decide to pass object name to update queries.